lib/src/api/reaction.dart

```diff
@@ -56,8 +62,9 @@ ReactionDisposer when(
   bool Function() predicate,
   void Function() effect, {
   String name,
+  ReactiveContext context,
 }) =>
-    createWhenReaction(predicate, effect, name: name);
+    createWhenReaction(context ?? mobxContext, predicate, effect, name: name);
```

Comment:
I think the context should go into the named argument list

Reply:
I think that will keep the signature consistent throughout

Reply:
I figured that `createWhenReaction` is an internal helper function, and as such the context should be a named argument. I've had trouble before in forgetting to pass this kind of stuff around if it's optional everywhere, so I prefer that just the public API's have it as an optional named argument.

Reply:
ok that makes sense